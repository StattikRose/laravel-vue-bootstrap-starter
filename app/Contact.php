<?php

namespace App;

use Carbon\Carbon;
use Illuminate\Database\Eloquent\Model;

class Contact extends Model
{
    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'name', 'company', 'email', 'phone', 'message',
    ];

    public function getCreatedAtAttribute($created_at): string
    {
        //return $created_at;
        return Carbon::parse($created_at)->format('F jS, Y h:i:s A');
    }
}
