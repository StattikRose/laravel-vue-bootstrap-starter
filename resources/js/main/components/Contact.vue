<template>
    <div class="mt-10">
        <h5>Example Component for &lt;Contact&gt;</h5>
        <form
            method="post"
            novalidate
            @submit="checkForm"
        >
            <!-- Email -->
            <div class="form-group">
                <label for="email">Email address</label>
                <input
                    id="email"
                    v-model="$v.form.email.$model"
                    type="email"
                    class="form-control"
                    :class="{ 'is-invalid': attemptSubmit && $v.form.email.$invalid }"
                    aria-describedby="emailHelp"
                    placeholder="Enter email"
                >
                <div
                    v-if="!$v.form.email.required && attemptSubmit"
                    class="invalid-feedback"
                >
                    This field is required
                </div>
                <div
                    v-if="!$v.form.email.email && attemptSubmit"
                    class="invalid-feedback"
                >
                    Enter a valid email
                </div>

                <small
                    id="emailHelp"
                    class="form-text text-muted"
                >We'll never share your email with anyone else.</small>
            </div>

            <button
                type="submit"
                class="btn btn-primary"
            >
                Submit
            </button>
        </form>
    </div>
</template>
<script>
import { required, email } from 'vuelidate/lib/validators'

export default {
    data () {
        return {
            attemptSubmit: false,
            form: {
                email: ''
            }
        }
    },
    validations: {
        form: {
            email: { required, email }
        }
    },
    methods: {
        checkForm: function (e) {
            e.preventDefault()
            this.attemptSubmit = true

            if (this.name && this.phone && this.company && this.email && this.message) {
                return true
            }

            this.errors = {}

            if (!this.email) {
                this.missingEmail = true
                this.errors.email = 'Please enter your email.'
            }
            if (!this.name) {
                this.errors.name = 'Please enter your name.'
            }
            if (!this.phone) {
                this.errors.phone = 'Please enter your phone.'
            }
            if (!this.company) {
                this.errors.company = 'Please enter your company.'
            }
            if (!this.message) {
                this.errors.message = 'Please provide some message.'
            }
        },
        onSubmit: function (e) {
            e.preventDefault()
            alert('here')
        }
    }
}
</script>
