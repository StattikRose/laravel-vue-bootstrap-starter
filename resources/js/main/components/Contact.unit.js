// resources/js/main/components/Contact.vue
import { createLocalVue, shallowMount } from '@vue/test-utils'
import Contact from './Contact.vue'
import Vuelidate from 'vuelidate'

describe('Contact.vue Component', () => {
    const localVue = createLocalVue()
    localVue.use(Vuelidate)

    let wrapper

    // creates a new wrapper after each test
    beforeEach(() => {
        wrapper = shallowMount(Contact, {
            localVue,
            mocks: {},
            stubs: {}
        })
    })

    // destroy the wrapper after each test
    afterEach(() => {
        wrapper.destroy()
    })

    test('Should mount without crashing', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Test clearForm()', () => {
        wrapper.setData({
            formData: {
                name: 'test',
                email: 'test'
            }
        })
        // run the calculation
        wrapper.vm.clearForm()

        expect(wrapper.vm.formData.email).toBe('')
    })

    /*
    test('Test calculateBusinessHydration()', () => {
        wrapper.setData({
            business_days: 5,
            business_employees: 5
        })
        // run the calculation
        wrapper.vm.calculateBusinessHydration()

        expect(wrapper.vm.total_month_gallons).toEqual(16.666666666666668)
        expect(Number.isInteger(wrapper.vm.total_bottles)).toBe(true)
    })

    test('Test selectDays()', () => {
        // run the selectDays
        wrapper.vm.selectDays(2)

        expect(wrapper.vm.business_days).toEqual(2)
    })

    test('Test selectType()', () => {
        // run the selectType
        wrapper.vm.selectType('home')

        expect(wrapper.vm.type).toBe('home')
    })

    test('Test calculate()', () => {
        const mockMethods = {
            calculateHomeHydration: jest.fn(),
            calculateBusinessHydration: jest.fn()
        }
        const wrapper = shallowMount(Contact, {
            localVue,
            mocks: {},
            stubs: {},
            methods: mockMethods
        })

        // run the calculate
        wrapper.vm.calculate()
        expect(wrapper.vm.calculated).toBe(true)
        expect(mockMethods.calculateBusinessHydration).toHaveBeenCalled()

        wrapper.setData({
            calculated: false,
            type: 'home'
        })
        wrapper.vm.calculate()
        expect(wrapper.vm.calculated).toBe(true)
        expect(mockMethods.calculateHomeHydration).toHaveBeenCalled()
    })

    test('the output after home click contains the adult, teen and child inputs', async () => {
        const button = wrapper.find('#home-button')
        button.trigger('click')
        await wrapper.vm.$nextTick()
        expect(wrapper.html()).toContain('<input name="home_adults" type="number"')
        expect(wrapper.html()).toContain('<input name="home_teenagers" type="number"')
        expect(wrapper.html()).toContain('<input name="home_kids" type="number" id="home_kids"')
    })

    test('the output after business click contains the number of employees and days open button', async () => {
        const button = wrapper.find('#business-button')
        button.trigger('click')
        await wrapper.vm.$nextTick().then(() => {
            expect(wrapper.html()).toContain('<input name="business_employees" type="number"')
            expect(wrapper.html()).toContain('<button type="button"')
        })
    })

    test('Test display()', async () => {
        // home tests
        wrapper.setData({
            type: 'home',
            home_adults: 1,
            home_kids: 1,
            home_teenagers: 1
        })
        wrapper.vm.calculate()

        await wrapper.vm.$nextTick()

        expect(wrapper.html()).toContain('3 five gallon bottles')
        expect(wrapper.html()).toContain('For home use')

        // business tests
        wrapper.setData({
            type: 'business',
            business_days: 1,
            business_employees: 1
        })
        wrapper.vm.calculate()

        await wrapper.vm.$nextTick()

        expect(wrapper.html()).toContain('1 five gallon bottle')
        expect(wrapper.html()).toContain('1 employee')
        expect(wrapper.html()).toContain('1 day')
        expect(wrapper.html()).toContain('For business use')

        wrapper.setData({
            type: 'business',
            business_days: 5,
            business_employees: 300
        })
        wrapper.vm.calculate()

        await wrapper.vm.$nextTick()
        expect(wrapper.html()).toContain('5 days')
        expect(wrapper.html()).toContain('300 employees')
        expect(wrapper.html()).toContain('recommend a bottle-free cooler')
    })**/
})
