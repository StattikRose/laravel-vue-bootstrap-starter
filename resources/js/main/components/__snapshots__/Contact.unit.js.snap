// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact.vue Component matches snapshot 1`] = `
<div class="mt-10">
  <h5>Example Component for &lt;Contact&gt;</h5>
  <form method="post" novalidate="novalidate">
    <div class="form-group"><label for="name">Name</label> <input id="name" type="text" placeholder="Enter name" class="form-control">
      <!---->
    </div>
    <div class="form-group"><label for="email">Email address</label> <input id="email" type="email" aria-describedby="emailHelp" placeholder="Enter email" class="form-control">
      <!---->
      <!----> <small id="emailHelp" class="form-text text-muted">We'll never share your email with anyone else.</small></div> <button type="submit" class="btn btn-primary">
      Submit
    </button>
    <!---->
    <!---->
    <!---->
  </form>
</div>
`;
